### ISSUE TITLE
Text box for file to be profiled accept directories

### ISSUE THREAD

**Issue by ccordoba12** [spyder-ide/spyder.line_profiler#25]
@Neotric commented on [Wed Dec 06 2017](https://github.com/spyder-ide/spyder/issues/5905)

## Description

**What steps will reproduce the problem?**

1. install sw with sudo -H pip install spyder-line-profiler
2. profile by line program in spyder:

```python
# -*- coding: utf-8 -*-
"""
Spyder Editor

This is a temporary script file.
"""
import math

@profile
def my_cube(a):
    return a*a*a

a = 0
for i in range(1000):
    a += math.sin(my_cube(i))
```

**What is the expected output? What do you see instead?**

No reaktion, but error.

**Please provide any additional information below**

  File "/usr/local/lib/python3.5/dist-packages/spyder_line_profiler/widgets/lineprofiler.py", line 242, in start
    self.process.setWorkingDirectory(wdir)
TypeError: QProcess.setWorkingDirectory(str): argument 1 has unexpected type 'bool'

## Version and main components

* Spyder Version: 3.2.4 
* Python Version: 3.5.2
* Qt Versions: 5.5.1, PyQt5 5.5.1 on Linux

## Dependencies
```
pyflakes >=0.6.0 :  1.6.0 (OK)
pycodestyle >=2.3:  2.3.1 (OK)
pygments >=2.0   :  2.2.0 (OK)
pandas >=0.13.1  :  None (NOK)
numpy >=1.7      :  None (NOK)
sphinx >=0.6.6   :  1.6.5 (OK)
rope >=0.9.4     :  0.10.7 (OK)
jedi >=0.9.0     :  0.11.0 (OK)
psutil >=0.3     :  3.4.2 (OK)
nbconvert >=4.0  :  5.3.1 (OK)
sympy >=0.7.3    :  None (NOK)
cython >=0.21    :  None (NOK)
qtconsole >=4.2.0:  4.3.1 (OK)
IPython >=4.0    :  6.2.1 (OK)
pylint >=0.25    :  1.7.4 (OK)
```

---

@ccordoba12 commented on [Wed Dec 06 2017](https://github.com/spyder-ide/spyder/issues/5905#issuecomment-349802585)

I'm moving this to its appropriate repo.

**Comment by ccordoba12** [comment]
@jitseniesen, does our line-profiler work with the latest Spyder version?

**Comment by jitseniesen** [comment]
Yes, the line-profiler plugin does work with the latest Spyder version (3.2.4). However, the error looks familiar; I think I saw it before.

@Neotric Your information is not enough for me to reproduce easily. Can you reproduce it yourself on your system? 

The error has something to do with paths. Could you please tell me the full paths of the Python file, and also the contents of the text field to the left of the `Profile by line` button in the line profiler?

**Comment by Neotric** [comment]
Yesterday, I downloaded anaconda (https://repo.continuum.io/archive/Anaconda3-5.0.1-Linux-x86.sh) and installed it. From anaconda I started Spyder. I was following an article in the 2017 nov. issue of Linux Journal.
It shows you to run the profiler and a 'better' profiler: spyder-line-profiler. Instruction to install that one does nor work:
$ conda install -c spyder-ide spyder-line-profiler
The package could not be found nor from within anaconda-navigator.
I could install with pip, but nevertheless it did not work properly
This morning I used the Linux Mint Software Manager and installed Spyder3 and Python3-spyderlib. Now the spyder-line-profiler is working.
I more than ofter find you can better wait for Mint or Ubuntu to add a package in their repo's. Then you end with less problems installing sw properly.

**Comment by jitseniesen** [comment]
Glad to hear that it now works for you.

It is possible to get that error if you set the text field with the file to be profiled to a directory instead of a Python file. This needs to be fixed.

**Comment by Neotric** [comment]
@jitseniesen, It ran OK when I did: Shift-F10, or started it from the menu. But when I hit the button "Profile by line" I got the error. Unfortunately, I can not unselect the file and select the folder. But after selecting the file (hit the folder icon near the 'Profile by line' button, it ran OK. But there remains a text "Profiling, please wait..." That only disappears after I click in the textbox. Than date and time appear.

By the way, why would the default profiler have difficulties with '@profile' in my code ? "NameError: name 'profile' is not defined".

**Comment by jitseniesen** [comment]
Actually, I just tried again (on a different computer) and now I also can't select the folder. I don't know what I did. But I can confirm what you say. To reproduce:
1. Open a file
2. Click `Profile line by line` in the `Run` menu; this works fine.
3. Click `Profile by line` in the Line profiler dock window; the error results.

> By the way, why would the default profiler have difficulties with '@profile' in my code ? "NameError: name 'profile' is not defined".

The `profile` decorator is defined in the line profiler but not in the standard profiler. You cannot use it in standard Python code either. There is a [question on StackOverflow](https://stackoverflow.com/questions/18229628/python-profiling-using-line-profiler-clever-way-to-remove-profile-statements) with a way to avoid this issue if you're interested (caveat: I didn't test it, or even read it).

**Comment by dalthviz** [comment]
Checking, seems like this is not an issue anymore I was not able to reproduce it:

![lineprof](https://user-images.githubusercontent.com/16781833/169092516-b20294a5-f12c-453a-82ec-66806f063dba.gif)

So should we close this one as fixed by #56 too @ccordoba12 @jitseniesen ?

**Comment by ccordoba12** [comment]
I think so. So many things have happened in our code base since this issue was reported, that I'm not surprised this is no longer a problem.
